src/sentry/static/sentry/app/components/events/interfaces/threads/threadSelector/threadSelector.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import {css} from '@emotion/core';
+
+import {Thread} from 'app/types/events';
+import {Event, EntryTypeData} from 'app/types';
+import DropdownAutoComplete from 'app/components/dropdownAutoComplete';
+import DropdownButton from 'app/components/dropdownButton';
+import theme from 'app/utils/theme';
+import {t} from 'app/locale';
+
+import {filterThreadInfo} from './filterThreadInfo';
+import {getThreadException} from './getThreadException';
+import {Option} from './option';
+import {SelectedOption} from './selectedOption';
+import {Header} from './header';
+
+type Props = {
+  threads: Array<Thread>;
+  activeThread: Thread;
+  event: Event;
+  onChange?: (thread: Thread) => void;
+};
+
+const DROPDOWN_MAX_HEIGHT = 400;
+
+const ThreadSelector = ({threads, event, activeThread, onChange}: Props) => {
+  const getDropDownItem = (thread: Thread) => {
+    const threadInfo = filterThreadInfo(thread, event);
+
+    const dropDownValue = `#${thread.id}: ${thread.name} ${threadInfo.label} ${threadInfo.filename}`;
+    let crashedInfo: undefined | EntryTypeData;
+
+    if (thread.crashed) {
+      crashedInfo = getThreadException(thread, event);
+    }
+
+    return {
+      value: dropDownValue,
+      threadInfo,
+      thread,
+      label: (
+        <Option
+          id={thread.id}
+          details={threadInfo}
+          name={thread.name}
+          crashed={thread.crashed}
+          crashedInfo={crashedInfo}
+        />
+      ),
+    };
+  };
+
+  const handleOnChange = ({thread}) => {
+    if (onChange) {
+      onChange(thread);
+    }
+  };
+
+  return (
+    <StyledDropdownAutoComplete
+      items={threads.map(getDropDownItem)}
+      onSelect={handleOnChange}
+      align="left"
+      alignMenu="left"
+      maxHeight={DROPDOWN_MAX_HEIGHT}
+      placeholder={t('Filter Threads')}
+      emptyMessage={t('You have no threads')}
+      noResultsMessage={t('No threads found')}
+      zIndex={theme.zIndex.dropdown}
+      css={css`
```

Comment:
Why do you need to use both a styled component and inline CSS?

Reply:
good catch 👏